packages/sdk/src/mint/mint.ts

```diff
@@ -36,80 +35,122 @@ export const mint = (
   }
 
   if (amount && amount > 125) {
     throw  new Error(ERROR_MESSAGES.MAX_AMOUT);
   }
 
-  if (royaltyPercentage && royaltyPercentage < 0 || royaltyPercentage > 0.5) {
-    throw new Error(ERROR_MESSAGES.INVALID_ROYALTY_PERCENTAGE);
+  //if specifying tokenIdsToMint these must be populated
+  if (tokenIdsToMint && tokenIdsToMint.length == 0) {
+    throw  new Error(ERROR_MESSAGES.EMPTY_TOKEN_IDS);
   }
 
+  if (tokenIdsToMint && amount && tokenIdsToMint?.length !== amount) {
+    throw new Error (ERROR_MESSAGES.MUTUAL_EXCLUSIVE_AMOUNT);
+  }
+  const adjustedTokenCount = tokenIdsToMint?.length ? tokenIdsToMint?.length : amount;
+  //if no amount or tokenids defined then default to 1 amount
+  const adjustedAmount = !amount && !tokenIdsToMint ? 1 : adjustedTokenCount; 
+  
+
   return {
     contractAddress: contractAddress || mbjs.keys.contractAddress,
     args: {
       owner_id: ownerId,
       metadata: metadata,
-      num_to_mint: amount || 1,
+      num_to_mint: adjustedAmount,
       // 10_000 = 100% (see above note)
-      royalty_args: !splits ? null : { split_between: splits, percentage: royaltyPercentage * 10_000 },
-      split_owners: splits || null,
-      token_ids_to_mint: tokenIdsToMint,
+      royalty_args: royaltyTotal < 0 || !royaltyTotal ? null : { split_between: roundedRoyalties, percentage: Math.floor(royaltyTotal * 10000) },
+      token_ids_to_mint: !tokenIdsToMint ? null : tokenIdsToMint,
     },
     methodName: TOKEN_METHOD_NAMES.MINT,
     gas: GAS,
     deposit: mintingDeposit({
-      nTokens: amount,
-      nRoyalties: !splits ? 0 : Object.keys(splits).length,
-      nSplits: splits ? Object.keys(splits).length : 0,
+      nTokens: adjustedTokenCount,
+      nRoyalties: !royalties ? 0 : Object.keys(royalties)?.length,
       metadata,
     }),
   };
 };
 
-function adjustSplitsForContract(splits: Record<string, number> ): void {
+function getRoyaltyTotal(royalties: Record<string, number> ): number {
+  let royaltyTotal = 0;
+  Object.values(royalties).forEach(value => {
+    royaltyTotal += value;
+  });
+
+  if (royaltyTotal <= 0 || royaltyTotal > 0.5) {
+    throw new Error (ERROR_MESSAGES.INVALID_ROYALTY_PERCENTAGE);
+  }
+  return royaltyTotal;  
+}
+
+function adjustRoyaltiesForContract(royalties: Record<string, number>, royaltyTotal ): Splits {
   let counter = 0;
-  Object.keys(splits).forEach(key => {
-    counter += splits[key];
-    splits[key] *= 10_000;
+  const result: Splits = {};
+  Object.keys(royalties).forEach(key => {
+    if (royalties[key] <= 0) {
+      throw new Error (ERROR_MESSAGES.NEGATIVE_ROYALTIES);
+    }
+    const adjustedAmount = royalties[key]/ royaltyTotal * 10000;
+    result[key] = adjustedAmount;
+    counter += adjustedAmount;
   });
-  if (counter != 1) {
+  if (counter != 10000) {
     throw new Error (ERROR_MESSAGES.SPLITS_PERCENTAGE);
   }
+ 
+  return result;
+}
+
+function roundRoyalties(royalties: Record<string, number>): Record<string, number> {
```

Comment:
Is this well-tested with automated tests?

Reply:
Yes, tried to cover some edgecases